src/main/java/io/omnition/loadgenerator/util/SummaryLogger.java

```diff
@@ -0,0 +1,56 @@
+package io.omnition.loadgenerator.util;
+
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Logger;
+
+/**
+ * SummaryLogger maintains global count of emitted traces and periodically
+ * outputs emitted counters and speed. Intended to be used as a single instance
+ * per app.
+ */
+public class SummaryLogger {
+    // How often to print summary log. 1 second.
+    private static final long SUM_LOG_PERIOD_NANOSEC = TimeUnit.NANOSECONDS.convert(1, TimeUnit.SECONDS);
+
+    // Last time summary log was printed, in monotinic nanoseconds
+    private long lastSumLogTimestampNano;
+
+    // Count of traces emited since summary log last printed
+    private long emitsSinceLastSumLog;
+    private long totalEmits;
+
+    private final Logger logger;
+
+    public SummaryLogger(Logger logger) {
+        this.logger = logger;
+    }
+
+    /**
+     * Count emitted traces and output summary if enough time elapsed since last
+     * output. This function is thread safe and can be called from multipler trace
+     * emitters concurrently.
+     */
+    public synchronized void logEmit(long emittedTraces) {
+        // Count emits
+        this.emitsSinceLastSumLog += emittedTraces;
+        this.totalEmits += emittedTraces;
+
+        // Check if it is time to output summary. Use nanoTime() as monotonic clock.
+        long curNanoTime = System.nanoTime();
```

Comment:
one problem with nanotime is that it is not safe across threads

Reply:
Do you mean it is not safe to call from multiple threads (as in it may corrupt a global counter) or calling from multiple threads may use different baseline counters and thus relative time is not comparable?
I cannot see anything mentioned in the docs, in fact docs make it sound like it should be fine:

> This method can only be used to measure elapsed time and is not related to any other notion of system or wall-clock time. The value returned represents nanoseconds since some fixed but arbitrary origin time (perhaps in the future, so values may be negative). The same origin is used by all invocations of this method in an instance of a Java virtual machine; other virtual machine instances are likely to use a different origin. 

Note: `The same origin is used`

Reply:
My understanding was that when different threads call System.nanoTime, the elapsed time duration can be inaccurate, and not super reliable across threads. Mayabe it changed in newer versions of java.

Reply:
It *does* seem to be thread safe after all: https://stackoverflow.com/questions/510462/is-system-nanotime-completely-useless/54566928#54566928

>Since Java 7, System.nanoTime() is guaranteed to be safe by JDK specification. System.nanoTime()'s Javadoc makes it clear that all observed invocations within a JVM (that is, across all threads) are monotonic